### PR TITLE
DEV: remove downstream test packages from environment.yml

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/issues/49998

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
Cool, seems to be working!

Looks good to me pending green (well, barring https://github.com/pandas-dev/pandas/issues/50124)

File: environment.yml [hunk 1]
```diff
@@ -60,17 +64,7 @@ dependencies:
 
   # downstream packages
   - aiobotocore<2.0.0  # GH#44311 pinned to fix docbuild
-  - botocore
-  - cftime
-  - dask
-  - ipython
-  - seaborn
```
Comment (MarcoGorelli) [inline environment.yml:67]: nvm, looks like we still need seaborn for now

I don't think it's a heavy dependency, let's just keep it in for now?

File: environment.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name: pandas-dev
 channels:
   - conda-forge
 dependencies:
-  - python=3.8
+  - python=3.10
```
Comment (MarcoGorelli) [inline environment.yml:6]: might be responsible for some typing failures? perhaps this change is best left to a separate pr

File: environment.yml (hunk 1 above)
Comment (jorisvandenbossche) [reply environment.yml:67]: Yeah, it's still used in the docs but just for a  colormap, so need to update that section to use a matplotlib built-in colormap.

File: environment.yml (hunk 2 above)
Comment (jorisvandenbossche) [reply environment.yml:6]: Ah, didn't think of that, yes will leave it for another PR then

File: environment.yml (hunk 1 above)
Comment (MarcoGorelli) [reply environment.yml:67]: shall we do that separately and remove the rest of the (much heavier) dependencies? seaborn isn't that big

Comment (jorisvandenbossche) [reply environment.yml:67]: I think I will just do that here, it's only a small change, and seaborn does bring in statsmodels

Comment (MarcoGorelli) [reply environment.yml:67]: no objections if it's a small change, though don't worry about it if it turns out to be more effort, looks like statsmodels is optional for seaborn (don't know if it was required in the past?)
```
(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ pip install seaborn
Collecting seaborn
  Using cached seaborn-0.12.1-py3-none-any.whl (288 kB)
Requirement already satisfied: pandas>=0.25 in ./.venv/lib/python3.8/site-packages (from seaborn) (1.4.4)
Requirement already satisfied: numpy>=1.17 in ./.venv/lib/python3.8/site-packages (from seaborn) (1.25.0.dev0+165.g6f9237e91)
Collecting matplotlib!=3.6.1,>=3.1
  Using cached matplotlib-3.6.2-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (9.4 MB)
Collecting pillow>=6.2.0
  Using cached Pillow-9.3.0-cp38-cp38-manylinux_2_28_x86_64.whl (3.3 MB)
Requirement already satisfied: python-dateutil>=2.7 in ./.venv/lib/python3.8/site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (2.8.2)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Requirement already satisfied: packaging>=20.0 in ./.venv/lib/python3.8/site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (21.3)
Requirement already satisfied: pyparsing>=2.2.1 in ./.venv/lib/python3.8/site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (3.0.9)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.38.0-py3-none-any.whl (965 kB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.4.4-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.2 MB)
Collecting contourpy>=1.0.1
  Using cached contourpy-1.0.6-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (295 kB)
Requirement already satisfied: pytz>=2020.1 in ./.venv/lib/python3.8/site-packages (from pandas>=0.25->seaborn) (2022.6)
Requirement already satisfied: six>=1.5 in ./.venv/lib/python3.8/site-packages (from python-dateutil>=2.7->matplotlib!=3.6.1,>=3.1->seaborn) (1.16.0)
Installing collected packages: pillow, kiwisolver, fonttools, cycler, contourpy, matplotlib, seaborn
Successfully installed contourpy-1.0.6 cycler-0.11.0 fonttools-4.38.0 kiwisolver-1.4.4 matplotlib-3.6.2 pillow-9.3.0 seaborn-0.12.1
(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ . repl.sh

In [1]: import statsmodels
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In [1], line 1
----> 1 import statsmodels

ModuleNotFoundError: No module named 'statsmodels'

```

**Comment by jorisvandenbossche** [comment]
There is still a doc error from using statsmodels in an old whatsnew file, will convert that to plain code block.

**Comment by WillAyd** [comment]
lgtm when green

**Comment by MarcoGorelli** [comment]
> There is still a doc error from using statsmodels in an old whatsnew file, will convert that to plain code block.

I went ahead and did this, hope it's OK

**Comment by jorisvandenbossche** [comment]
> I went ahead and did this, hope it's OK

Sure!

**Comment by jorisvandenbossche** [comment]
Should we document in the contributing docs for now that you need to install a few additional packages to be able to run all tests in `test_downstream.py` ?  
(`pip/conda install seaborn scikit-learn statsmodels cftime pandas-datareader`)

I am not fully sure this is necessarily worth mentioning (it makes the docs more complex with yet another note, while in practice contributors won't often need this)

**Comment by MarcoGorelli** [comment]
I think it's OK to not mention this - very few contributors will be running the downstream tests anyway, and for those who do, the error message will typically make it clear what they need to install

**Comment by jorisvandenbossche** [comment]
I additionally also removed `pandas-gbq`, because 1) this is actually a big dependency (with conda, indirectly it is still bringing in geopandas and its dependencies (GDAL etc)), and 2) we actually only need this for `test_downstream.py` so it _is_  a "downstream" dependency in that sense (it seems that all actual tests of `read_gbq` / `to_gbq` have been removed because this functionality is tested in pandas-gbq itself now)

**Comment by phofl** [comment]
It would be nice to have a list of downstream dependencies somewhere. Not necessarily in the docs though. Maybe inside the test file itself?

**Review by mroeschke — APPROVED** [review]
LGTM (IMO fine either way documenting these need to be installed or not)

**Pull request merged**